c2rust-analyze/src/main.rs

```diff
@@ -800,97 +800,115 @@ fn run(tcx: TyCtxt) {
         );
     }
 
-    // Before generating rewrites, add the FIXED flag to the signatures of all functions that
-    // failed analysis.
-    for did in gacx.iter_fns_failed() {
-        let lsig = gacx.fn_sigs[&did];
-        for sig_lty in lsig.inputs_and_output() {
-            for lty in sig_lty.iter() {
-                let ptr = lty.label;
-                if !ptr.is_none() {
-                    gasn.flags[ptr].insert(FlagSet::FIXED);
-                }
-            }
-        }
-    }
-
     // Buffer debug output for each function.  Grouping together all the different types of info
     // for a single function makes FileCheck tests easier to write.
     let mut func_reports = HashMap::<LocalDefId, String>::new();
 
     // Generate rewrites for all functions.
     let mut all_rewrites = Vec::new();
-    for &ldid in &all_fn_ldids {
-        if gacx.fn_failed(ldid.to_def_id()) {
-            continue;
+
+    // It may take multiple tries to reach a state where all rewrites succeed.
+    loop {
+        func_reports.clear();
+        all_rewrites.clear();
+        eprintln!("\n\n ------ START LOOP ------");
```

Comment:
nit: I think something like `START REWRITE LOOP` would be better

Reply:
I cleaned up the debug prints a little bit